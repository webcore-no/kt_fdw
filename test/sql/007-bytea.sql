set client_min_messages=ERROR;
drop server if exists kt_server cascade;
set client_min_messages=NOTICE;


create server kt_server foreign data wrapper kt_fdw;
create foreign table kt_test (key bytea, value bytea, flag text) server kt_server;
create user mapping for public server kt_server;
delete from kt_test;
insert into kt_test values (E'\\xDEAD00BEEF'::bytea, E'\\xDEAD00BEEF'::bytea);
select * from kt_test;
insert into kt_test values (E'\\xDEAD00BEEF'::bytea, 'foo', 'kt_set');
select * from kt_test;
update kt_test set value = 'bar' where key = E'\\xDEAD00BEEF'::bytea;
select * from kt_test;
delete from kt_test where key = E'\\xDEAD00BEEF'::bytea;
select * from kt_test;
