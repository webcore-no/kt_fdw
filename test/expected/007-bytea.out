set client_min_messages=ERROR;
drop server if exists kt_server cascade;
set client_min_messages=NOTICE;
create server kt_server foreign data wrapper kt_fdw;
create foreign table kt_test (key bytea, value bytea, flag text) server kt_server;
create user mapping for public server kt_server;
delete from kt_test;
insert into kt_test values (E'\\xDEAD00BEEF'::bytea, E'\\xDEAD00BEEF'::bytea);
select * from kt_test;
     key      |    value     | flag 
--------------+--------------+------
 \xdead00beef | \xdead00beef | 
(1 row)

insert into kt_test values (E'\\xDEAD00BEEF'::bytea, 'foo', 'kt_set');
select * from kt_test;
     key      |  value   | flag 
--------------+----------+------
 \xdead00beef | \x666f6f | 
(1 row)

update kt_test set value = 'bar' where key = E'\\xDEAD00BEEF'::bytea;
select * from kt_test;
     key      |  value   | flag 
--------------+----------+------
 \xdead00beef | \x626172 | 
(1 row)

delete from kt_test where key = E'\\xDEAD00BEEF'::bytea;
select * from kt_test;
 key | value | flag 
-----+-------+------
(0 rows)

